core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -98,86 +93,90 @@ object StaticFile {
                 Some(
                   Response(
                     headers = headers,
-                    body = readInputStream[F](F.pure(inputStream), DefaultBufferSize, blocker)
+                    body = readInputStream[F](F.pure(inputStream), DefaultBufferSize)
                   ))
               }
             )
         } else
-          blocker
-            .delay(urlConn.getInputStream.close())
+          F.blocking(urlConn.getInputStream.close())
             .handleError(_ => ())
             .as(Some(Response(NotModified)))
       }
     })
   }
 
-  def calcETag[F[_]: Sync]: File => F[String] =
+  def calcETag[F[_]: Files: MonadError[*[_], Throwable]]: File => F[String] =
```

Comment:
Why not just Functor here?

Reply:
I must've mistakenly replaced it along with some other `Functors` with `MonadError` while testing the waters. I'll get it changed to `Functor`.